src/components/GlobalPreferences/Notifications/Notifications.js

```diff
@@ -45,16 +67,9 @@ function useAuthState() {
     }
 
     if (email && token) {
-      setAuthState(AUTH_AUTHENTICATING)
-      isAuthTokenValid(token)
-        .then(v => {
-          setAuthState(AUTH_AUTHENTICATED)
-          return v
-        })
-        .catch(e => {
-          setAuthState(AUTH_AUTHENTICATION_FAILED)
-        })
+      handleAuthenticate()
     }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [email, token])
```

Comment:
Shouldn't this include `handleAuthenticate()`? Otherwise if `token` changes, the local `handleAuthenticate()` might not use the new `token`.